### PR TITLE
Modifies config files to provide more default values and updates README

### PR DESCRIPTION
**Pull request by saltiyazan**
Modifies config files to provide more default values and updates README

### CODE REVIEW

**Review by ghislainbourgeois — APPROVED** [review]

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -73,9 +79,10 @@ parts:
...
       ./usr/share/srsran/rb.conf.example: ./usr/share/srsran/rb.conf
-    override-build: |
-      set -x
-      pushd $SNAPCRAFT_PART_SRC
-      sed -ie 's/native/x86_64/g' CMakeLists.txt
-      popd
-      snapcraftctl build
+    override-prime: |
+      snapcraftctl prime
+      sed -i '/device_args = tx_port=tcp/s/^#//g' ./usr/share/srsran/ue.conf
```
Comment (gruyaume) [inline snap/snapcraft.yaml:84]: I'm not a huge fan of those `sed` command as they require us to assume the content of the config file. I'm fine if we have the complete config files here.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -73,9 +79,10 @@ parts:
...
+      sed -i '/device_name = zmq/s/^#//g' ./usr/share/srsran/ue.conf
+      sed -i '/device_args = fail_on_disconnect/s/^#//g' ./usr/share/srsran/enb.conf
+      sed -i '/device_name = zmq/s/^#//g' ./usr/share/srsran/enb.conf
+
```
Comment (gruyaume) [inline snap/snapcraft.yaml:88]: I know this is somewhat unrelated to this PR but we should probably change the default log file as well

File: snap/snapcraft.yaml (hunk 1 above)
Comment (saltiyazan) [reply snap/snapcraft.yaml:84]: Yes, I also thought of this, but we are using a specific release tag, not the latest version of the code.
So maybe we can assume it won't change.
I'm happy to take it out to the README instead, let me know what you think :)

File: snap/snapcraft.yaml (hunk 2 above)
Comment (saltiyazan) [reply snap/snapcraft.yaml:88]: Do you have a suggestion where to log? :) we can include it here, no problem.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (gruyaume) [reply snap/snapcraft.yaml:84]: This assumption is somewhat applicable to both the `sed` and the complete config file. It's more that f we provide the config file we **know** what we are publishing. Here we were doing a `sed` on a file that we don't know the content of.

Comment (saltiyazan) [reply snap/snapcraft.yaml]: Good point. Done.

File: snap/snapcraft.yaml (hunk 2 above)
Comment (saltiyazan) [reply snap/snapcraft.yaml]: It will log to  `/var/snap/srsran/current/` by default.
Done.

File: snap/local/enb.conf [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+# bind_port: Bind port for MAC network trace (default: 5687)
+# client_ip: Client IP address for MAC network trace (default: "127.0.0.1")
+# client_port Client IP address for MAC network trace (default: 5847)
+#####################################################################
+[pcap]
+#enable = false
+#filename = /tmp/enb_mac.pcap
+#nr_filename = /tmp/enb_mac_nr.pcap
+#s1ap_enable = false
+#s1ap_filename = /tmp/enb_s1ap.pcap
```
Comment (ghislainbourgeois) [inline snap/local/enb.conf:126]: Did you try to use those captures? I think the snap cannot directly write to those locations, so we could change the default here to something that will work with the snap.

File: snap/local/epc.conf [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+# To use the dissector, edit the preferences for DLT_USER to 
+# add an entry with DLT=150, Payload Protocol=s1ap.
+#
+# enable:   Enable or disable the PCAP.
+# filename: File name where to save the PCAP.
+#
+####################################################################
+[pcap]
+enable   = false
+filename = /tmp/epc.pcap
```
Comment (ghislainbourgeois) [inline snap/local/epc.conf:81]: Same thing here

File: snap/local/ue.conf [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+# enable:            Enable packet captures of layers (mac/mac_nr/nas/none) multiple option list
+# mac_filename:      File path to use for MAC packet capture
+# mac_nr_filename:   File path to use for MAC NR packet capture
+# nas_filename:      File path to use for NAS packet capture
+#####################################################################
+[pcap]
+enable = none
+mac_filename = /tmp/ue_mac.pcap
+mac_nr_filename = /tmp/ue_mac_nr.pcap
+nas_filename = /tmp/ue_nas.pcap
```
Comment (ghislainbourgeois) [inline snap/local/ue.conf:97]: Same thing here

File: snap/local/ue.conf [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+[general]
+#metrics_csv_enable    = false
+#metrics_period_secs   = 1
+#metrics_csv_filename  = /tmp/ue_metrics.csv
+#have_tti_time_stats   = true
+#tracing_enable        = true
+#tracing_filename      = /tmp/ue_tracing.log
+#tracing_buffcapacity  = 1000000
+#metrics_json_enable   = false
+#metrics_json_filename = /tmp/ue_metrics.json
```
Comment (ghislainbourgeois) [inline snap/local/ue.conf:467]: Same thing for those files

**Review by ghislainbourgeois — APPROVED** [review]

File: snap/local/enb.conf (hunk 3 above)
Comment (saltiyazan) [reply snap/local/enb.conf:126]: No, I didn't, but this is commented out, so to enable it the users would need to modify the config themself.
Do you suggest to support default values?

File: snap/local/ue.conf (hunk 6 above)
Comment (saltiyazan) [reply snap/local/ue.conf:467]: same here, that they are disabled.

Comment (saltiyazan) [reply snap/local/ue.conf]: also updated
Done

File: snap/local/ue.conf (hunk 5 above)
Comment (saltiyazan) [reply snap/local/ue.conf]: Done.

File: snap/local/epc.conf (hunk 4 above)
Comment (saltiyazan) [reply snap/local/epc.conf]: This is disabled by default as above, the difference that the user can enable it without the need to modify the filename config.
We could save those files to the data dir of the snap. like the logs.
Done.

File: snap/local/enb.conf (hunk 3 above)
Comment (saltiyazan) [reply snap/local/enb.conf]: However I updated it to by in synch with other files where it is uncommented by default.
done

**Review by gruyaume — APPROVED** [review]

File: snap/local/enb.conf [hunk 7]
```diff
@@ -0,0 +1,429 @@
...
+# bind_port: Bind port for MAC network trace (default: 5687)
+# client_ip: Client IP address for MAC network trace (default: "127.0.0.1")
+# client_port Client IP address for MAC network trace (default: 5847)
+#####################################################################
+[pcap]
+#enable = false
+#filename = /var/snap/srsran/current/enb_mac.pcap
+#nr_filename = /var/snap/srsran/current/enb_mac_nr.pcap
+#s1ap_enable = false
+#s1ap_filename = /var/snap/srsran/current/enb_s1ap.pcap
```
Comment (ghislainbourgeois) [inline snap/local/enb.conf:126]: I think the change will not take effect if the lines stay commented out. I would uncomment all the `filename` lines where we change a default.

File: snap/local/enb.conf [hunk 8]
```diff
@@ -0,0 +1,429 @@
...
+#metrics_period_secs  = 1
+#metrics_csv_enable   = false
+#metrics_csv_filename = /tmp/enb_metrics.csv
+#report_json_enable   = true
+#report_json_filename = /tmp/enb_report.json
+#report_json_asn1_oct = false
+#alarms_log_enable    = true
+#alarms_filename      = /tmp/enb_alarms.log
+#tracing_enable       = true
+#tracing_filename     = /tmp/enb_tracing.log
```
Comment (ghislainbourgeois) [inline snap/local/enb.conf:407]: Some more default paths that will not work

File: snap/local/ue.conf [hunk 9]
```diff
@@ -0,0 +1,467 @@
...
+[general]
+#metrics_csv_enable    = false
+#metrics_period_secs   = 1
+#metrics_csv_filename  = /var/snap/srsran/current/ue_metrics.csv
+#have_tti_time_stats   = true
+#tracing_enable        = true
+#tracing_filename      = /var/snap/srsran/current/ue_tracing.log
+#tracing_buffcapacity  = 1000000
+#metrics_json_enable   = false
+#metrics_json_filename = /var/snap/srsran/current/ue_metrics.json
```
Comment (ghislainbourgeois) [inline snap/local/ue.conf:467]: I would uncomment those `filename` lines as well.

File: snap/local/enb.conf (hunk 3 above)
Comment (saltiyazan) [reply snap/local/enb.conf]: True.

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: snap/local/enb.conf (hunk 7 above)
Comment (saltiyazan) [reply snap/local/enb.conf]: Fixed

File: snap/local/enb.conf (hunk 8 above)
Comment (saltiyazan) [reply snap/local/enb.conf]: Fixed.

File: snap/local/ue.conf (hunk 9 above)
Comment (saltiyazan) [reply snap/local/ue.conf]: Fixed

**Review by ghislainbourgeois — APPROVED** [review]

**Comment by saltiyazan** [comment]
> Please make sure the CI issues are fixed before merging

It passed, I will merge it :)

**Pull request merged**